components/ProgressIndicator/ProgressIndicator.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { number } from 'prop-types';
+import styles from './ProgressIndicator.css';
+
+ProgressIndicator.propTypes = {
+  stepNumber: number,
+  totalSteps: number,
+};
+
+ProgressIndicator.defaultProps = {
+  stepNumber: 0,
+  totalSteps: 0,
+};
+
+export default function ProgressIndicator({ stepNumber, totalSteps }) {
+  if (totalSteps < 1) return null;
+
+  let currentStep = stepNumber;
```

Comment:
Instead of this, let's just yell at the developer when they do something silly:

```jsx
// exported so you can easily import this into test files to test that the error is thrown
export const developerErrors = {
  currentStepTooLow: "You can't have a negative current step",
  currentStepTooHigh:
    "You can't have a current stop that's greater than the total number of steps",
};

function ProgressIndicator({ stepNumber, totalSteps }) {
// stuff
  if (process.env.NODE_ENV !== 'production') {
    if (currentStep < 0) {
      throw new Error(developerErrors.currentStepTooLow);
    }

    if (currentStep > totalSteps) {
      throw new Error(developerErrors.currentStepTooHigh);
    }
  }

  // stuff
}

export default ProgressIndicator;
```

Reply:
pushed!